### PR TITLE
Add java support

### PR DESCRIPTION
**Pull request by PitcherTear22**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -166,7 +166,7 @@ List of currently supported languages:
 - [x] html (maintained by @TravonteD)
 - [ ] csharp
 - [ ] swift
-- [ ] java
+- [x] java
```
Comment (vigoux) [inline README.md:169]: Please add your name here if you want to maintain this language.

File: queries/java/highlights.scm [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+(this) @variable.builtin
+
+; Literals
+
+(hex_integer_literal) @number
+(decimal_integer_literal) @number
+(octal_integer_literal) @number
+(decimal_floating_point_literal) @number
+(hex_floating_point_literal) @number
```
Comment (vigoux) [inline queries/java/highlights.scm:85]: These should be marked as `@float`

File: queries/java/highlights.scm [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+(character_literal) @string
```
Comment (vigoux) [inline queries/java/highlights.scm:86]: This one should be marked as `@character`

File: README.md (hunk 1 above)
Comment (PitcherTear22) [reply README.md:169]: Maybe later? After August?

**Comment by PitcherTear22** [comment]
> Thanks for your contribution, I assume that you don't want to be mentionned in the `README` ?

Not as maintainer. I can maintain after August

**Pull request merged**